### PR TITLE
Raise error if diff file is empty in interdiff mode

### PR DESCRIPTION
**Pull request by sofiia-tesliuk**

### CODE REVIEW

File: patchutils.go [hunk 1]
```diff
@@ -23,12 +23,20 @@ func InterDiff(oldDiff, newDiff io.Reader) (string, error) {
...
+
 	newFileDiffs, err := diff.NewMultiFileDiffReader(newDiff).ReadAllFiles()
 
 	if err != nil {
 		return "", fmt.Errorf("parsing newDiff: %w", err)
 	}
 
+	if len(newFileDiffs) == 0 {
+		return "", fmt.Errorf("newDiff: %w", ErrEmptyDiffFile)
+	}
```
Comment (schachmat) [inline patchutils.go]:
```suggestion
	newFileDiffs, err := diff.NewMultiFileDiffReader(newDiff).ReadAllFiles()
	if err != nil {
		return "", fmt.Errorf("parsing newDiff: %w", err)
	}
	if len(newFileDiffs) == 0 {
		return "", fmt.Errorf("newDiff: %w", ErrEmptyDiffFile)
	}
```

same for the block above.

File: patchutils_test.go [hunk 2]
```diff
@@ -16,16 +17,33 @@ var interDiffFileTests = []struct {
...
+		resultFile: "s2_a_b.diff",
+		wantErr:    ErrEmptyDiffFile,
+	},
+	{
+		diffAFile: "s2_a.diff",
+		// Not a diff file
+		diffBFile:  "source_1/file_2.txt",
+		resultFile: "s2_a_b.diff",
+		wantErr:    ErrEmptyDiffFile,
 	},
```
Comment (schachmat) [inline patchutils_test.go:47]: nit: add a test case with an empty diff file.

**Review by schachmat — CHANGES_REQUESTED** [review]

**Review by schachmat — APPROVED** [review]
